target_chains/solana/programs/pyth-solana-receiver/src/lib.rs

```diff
@@ -91,72 +102,79 @@ pub mod pyth_solana_receiver {
         // TODO: update pythnet_sdk to implement BorshSerialize, BorshDeserialize
         // for MerklePriceUpdate as well as Keccak160 price_updates can be passed
         // in as Vec<MerklePriceUpdate>
-        price_updates: Vec<Vec<u8>>,
+        price_update: MerklePriceUpdate,
     ) -> Result<()> {
-        let unchecked_vaa = &ctx.accounts.posted_vaa;
-        require_keys_eq!(
-            *unchecked_vaa.owner,
-            //TODO: expected owner should come from config account that can only be modified by governance
-            wormhole_anchor_sdk::wormhole::program::id(),
-            ReceiverError::WrongVaaOwner
-        );
-        let vaa = AnchorVaa::try_deserialize(&mut &**(unchecked_vaa.try_borrow_data()?))
-            .map_err(|_| ReceiverError::DeserializeVaaFailed)?;
-
-        // TODO: expected emitter_chain should come from config account that can only be modified by governance
-        require_eq!(
-            vaa.emitter_chain(),
-            <wormhole::Chain as Into<u16>>::into(wormhole::Chain::Pythnet),
-            ReceiverError::InvalidEmitterChain
-        );
-
-        require_keys_eq!(
-            Pubkey::new_from_array(*vaa.emitter_address()),
-            // TODO: expected emitter_address should come from config account that can only be modified by governance
-            Pubkey::new_from_array(ACCUMULATOR_EMITTER_ADDRESS),
-            ReceiverError::InvalidEmitterAddress
+        let config = &ctx.accounts.config;
+        let payer = &ctx.accounts.payer;
+        let encoded_vaa = &ctx.accounts.encoded_vaa;
+        let treasury = &ctx.accounts.treasury;
+        let price_update_account = &mut ctx.accounts.price_update_account;
+
+        if payer.lamports()
+            < Rent::get()?
+                .minimum_balance(0)
+                .saturating_add(config.single_update_fee_in_lamports)
+        {
+            return err!(ReceiverError::InsufficientFunds);
+        };
+
+        let transfer_instruction = system_instruction::transfer(
+            payer.key,
+            treasury.key,
+            config.single_update_fee_in_lamports,
         );
+        anchor_lang::solana_program::program::invoke(
+            &transfer_instruction,
+            &[
+                ctx.accounts.payer.to_account_info(),
+                ctx.accounts.treasury.to_account_info(),
+            ],
+        )?;
+
+        let (header, body): (Header, Body<&RawMessage>) =
```

Comment:
for my own edification: wormhole guarantees that any account owned by wormhole that can be deserialized using this function is a VAA? (i.e., the header discriminates between the different types of wormhole accounts)

Reply:
Wormhole guarantees that any account :
- Owned by wormhole
- With the right Anchor discriminator 
Is an EncodedVaa account.
Additionally if encoded_vaa.header.verified_signatures = n > 0, it means n guardian signatures have been verfied